examples/ex_mnist_transfer_puretf.py

```diff
@@ -0,0 +1,407 @@
+                   test_end=10000, nb_epochs=6, batch_size=128,
+                   learning_rate=0.1):
+    """
+    MNIST cleverhans tutorial
+    :param train_start: index of first training set example
+    :param train_end: index of last training set example
+    :param test_start: index of first test set example
+    :param test_end: index of last test set example
+    :param nb_epochs: number of epochs to train model
+    :param batch_size: size of training batches
+    :param learning_rate: learning rate for training
+    returns None
+    """
+
+    # Set TF random seed to improve reproducibility
+    tf.set_random_seed(1234)
+
+    # Create TF session
+    sess = tf.Session()
+
+    # Get MNIST test data
+    X_train, Y_train, X_test, Y_test = data_mnist(train_start=train_start,
+                                                  train_end=train_end,
+                                                  test_start=test_start,
+                                                  test_end=test_end)
+
+    # Use label smoothing
+    assert Y_train.shape[1] == 10.
+    label_smooth = .1
+    Y_train = Y_train.clip(label_smooth / 9., 1. - label_smooth)
+
+    # Define input TF placeholder
+    x = tf.placeholder(tf.float32, shape=(None, 28, 28, 1))
+    y = tf.placeholder(tf.float32, shape=(None, 10))
+
+    # Define TF model graph
+    model = make_basic_cnn()
+    preds = model.fprop(x)
+    print("Defined TensorFlow model graph.")
+
+    def evaluate():
+        # Evaluate the accuracy of the MNIST model on legitimate test examples
+        eval_params = {'batch_size': batch_size}
+        acc = model_eval(sess, x, y, preds, X_test, Y_test, args=eval_params)
+        assert X_test.shape[0] == test_end - test_start, X_test.shape
+        print('Test accuracy on legitimate examples: %0.4f' % acc)
+
+    # Train an MNIST model
+    train_params = {
+        'nb_epochs': nb_epochs,
+        'batch_size': batch_size,
+        'learning_rate': learning_rate
+    }
+    model_train(sess, x, y, preds, X_train, Y_train, evaluate=evaluate,
+                args=train_params)
+    
+    # Print initial confusion matrix, without adversarial samples
+    confusion(sess, x, y, preds, X_test, Y_test, batch_size=batch_size)
+    print("The above confusion matrix shows that after training, very", 
+          " few samples are misclassified.")
+
+    # Initialize the Fast Gradient Sign Method (FGSM) attack object and graph
+    fgsm = FastGradientMethod(model, sess=sess)
+    fgsm_params = {'eps': 0.3}
+    adv_x = fgsm.generate(x, **fgsm_params)
+    preds_adv = model.fprop(adv_x)
+
+    # Evaluate the accuracy of the MNIST model on adversarial examples
+    eval_par = {'batch_size': batch_size}
+    acc = model_eval(sess, x, y, preds_adv, X_test, Y_test, args=eval_par)
+    print('Test accuracy on adversarial examples: %0.4f\n' % acc)
+    
+    # Print confusion Matrix
+    confusion(sess, x, y, preds_adv, X_test, Y_test, batch_size=batch_size)
```

Comment:
Right now, when running the script it is somewhat hard to understand to which pair of models a confusion matrix refers to. Could you add a print that says something like **Confusion matrix for adversarial examples generated and evaluated on a CNN with 2 layers** and then for a transferability confusion matrix **Confusion matrix for adversarial examples generated on a CNN with 2 layers and evaluated on a CNN with 4 layers**

The number of layers are not exact but it should just indicate to the user what is the high-level difference between the two architectures.